packages/functions/src/dataTriggers.ts

```diff
@@ -269,7 +269,82 @@ export const createAttendanceForBooking = functions
 
     await attendanceRef.set(updatedEntry, { merge: true });
   });
+// SlotAttendnace = {
+//   /**
+//    * ISO date of the slot for easier querying
+//    */
+//   date: string;
+//   /**
+//    * Attendances for slot keyed by customer
+//    */
+//   attendances: {
+//     [customerId: string]: CustomerAttendance;
+//   };
+// };
+
+// CustomerAttendance {
+//   bookedInterval: string | null;
+//   attendedInterval: string | null;
+// }
+
+// entry
+// [slotId: string]: CustomerBookingEntry;
+// {
+//   /**
+//    * ISO date of the booked slot (used for easier querying from store)
+//    */
+//   date: string;
+//   /**
+//    * Booked interval for particular slot
+//    */
+//   interval: string;
+
+// }
+export const createAttendedSlotOnAttendance = functions
+  .region(__functionsZone__)
+  .firestore.document(
+    `${Collection.Organizations}/{organization}/${OrgSubCollection.Attendance}/{bookingId}/`
+  )
+  .onWrite(async (change, context) => {
+    const { organization, bookingId } = context.params as Record<
+      string,
+      string
+    >;
+
+    const db = admin.firestore();
+
+    const isCreate = !change.before.exists;
+    // const isDelete = !change.after.exists;
+
+    if (isCreate) {
+      const attendanceData = change.after.data() as SlotAttendnace;
+
+      const customerId = Object.keys(attendanceData.attendances)[0];
```

Comment:
This would update only the attended slots for the first customer attended (or booked) that day, as the attendance contains the following record:
```typescript
{
  [slotId: string]: {
    attendances: {
      [customerId: string]: {
        bookedInterval: string | null
        attendedInterval: string | null
      }
    }
  }
}
```

Reply:
You'd want to loop over all of the customers and update their attendances (but only if needed)